internal/exec/stages/disks/luks.go

```diff
@@ -85,140 +90,182 @@ func (s *stage) createLuks(config types.Config) error {
 	s.Logger.PushPrefix("createLuks")
 	defer s.Logger.PopPrefix()
 
-	for _, luks := range config.Storage.Luks {
-		// TODO: allow Ignition generated KeyFiles for
-		// non-clevis devices that can be persisted.
-		// TODO: create devices in parallel.
-		// track whether Ignition creates the KeyFile
-		// so that it can be removed
-		var ignitionCreatedKeyFile bool
-		// create keyfile inside of tmpfs, it will be copied to the
-		// sysroot by the files stage
-		os.MkdirAll(distro.LuksInitramfsKeyFilePath(), 0700)
-		keyFilePath := filepath.Join(distro.LuksInitramfsKeyFilePath(), luks.Name)
-		if util.NilOrEmpty(luks.KeyFile.Source) {
-			// create a keyfile
-			key, err := randHex(4096)
-			if err != nil {
-				return fmt.Errorf("generating keyfile: %v", err)
-			}
-			if err := ioutil.WriteFile(keyFilePath, []byte(key), 0400); err != nil {
-				return fmt.Errorf("creating keyfile: %v", err)
-			}
-			ignitionCreatedKeyFile = true
-		} else {
-			f := types.File{
-				Node: types.Node{
-					Path: keyFilePath,
-				},
-				FileEmbedded1: types.FileEmbedded1{
-					Contents: luks.KeyFile,
-				},
-			}
-			fetchOps, err := s.Util.PrepareFetches(s.Util.Logger, f)
-			if err != nil {
-				return fmt.Errorf("failed to resolve keyfile %q: %v", f.Path, err)
-			}
-			for _, op := range fetchOps {
-				if err := s.Util.Logger.LogOp(
-					func() error {
-						return s.Util.PerformFetch(op)
-					}, "writing file %q", f.Path,
-				); err != nil {
-					return fmt.Errorf("failed to create keyfile %q: %v", op.Node.Path, err)
-				}
+	// Create LUKS devices concurrently up to the lowest of
+	// GOMAXPROCS or the amount of available RAM in GB - 1
+	// (from testing LUKS devices needed up to ~1GB of memory
+	// per thread to avoid oomkill)
+	maxprocs := runtime.GOMAXPROCS(-1)
+	var sysinfo syscall.Sysinfo_t
+	err := syscall.Sysinfo(&sysinfo)
+	if err != nil {
+		return fmt.Errorf("determining system memory: %v", err)
+	}
+	memory := sysinfo.Totalram / 1024 / 1024 / 1024
```

Comment:
The compiler may do this, but I think it's cheaper to compute `sysinfo.Totalram / (1024 * 1024 * 1024)`.  Also I tend to add units to variable names for things like this, i.e. `memoryGiB := ...`.